src/smpmap.cpp

```diff
@@ -135,6 +135,19 @@ void SMPMap::update(const mcd::ClientboundMultiBlockChange& packet) {
       packet.chunkCoordinates.y, packet.records);
 }
 
+void SMPMap::update(const mcd::ClientboundBlockChange& packet) {
+  std::unique_lock lock(mutex);
+  if (chunks[{packet.location.x >> 4, packet.location.z >> 4}].sections[packet.location.y >> 4])
```

Comment:
What happens if this is getting placed into a chunk section that hasn't been created yet? It's good to check, but the chunk section needs to be `.emplaced()` if it doesn't already exist. Ideally this should happen inside the relevant `ChunkColumn`, not `SMPMap`.